crates/walrus-core/src/encoding/slivers.rs

```diff
@@ -115,17 +115,28 @@ impl<T: EncodingAxis> SliverData<T> {
             Ok(self.symbols.symbol_size()) == metadata.symbol_size(encoding_config),
             SliverVerificationError::SymbolSizeMismatch
         );
+        self.check_hash(&encoding_config_for_type, metadata)
+    }
+
+    /// Checks that the hash of the sliver matches the hash in the metadata.
+    ///
+    /// This assumes that all relevant size checks have already been performed.
+    pub fn check_hash(
```

Comment:
Nit: since the function has strict-ish assumptions on the inputs, and it is only used withing `walrus-core`, maybe it's worth restricting access?

```suggestion
    pub(crate) fn check_hash(
```